scikit/notebooks/Scikit Learn on CMLE.ipynb

```diff
@@ -0,0 +1,381 @@
+    "    data[i].append(int(np.uint64(test_features[col][i]).item()))\n",
+    "\n",
+    "# Write the test data to a json file\n",
+    "with open('data.json', 'w') as outfile:\n",
+    "  json.dump(data, outfile)\n",
+    "\n",
+    "# Get one person that makes <50K and one that makes >50K to test our model.\n",
+    "print('Show a person that makes <50K:')\n",
+    "print('\\tFeatures: {0} --> Label: {1}\\n'.format(data[0], test_labels[0]))\n",
+    "\n",
+    "with open('less_than_50K.json', 'w') as outfile:\n",
+    "  json.dump(data[0], outfile)\n",
+    "\n",
+    "  \n",
+    "print('Show a person that makes >50K:')\n",
+    "print('\\tFeatures: {0} --> Label: {1}'.format(data[2], test_labels[2]))\n",
+    "\n",
+    "with open('more_than_50K.json', 'w') as outfile:\n",
+    "  json.dump(data[2], outfile)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Use Gcloud to make online predictions\n",
+    "Use the two people (as seen in the table) gathered in the previous step for the gcloud predictions.\n",
+    "\n",
+    "| **Person** | age | workclass | fnlwgt | education | education-num | marital-status | occupation | relationship | race | sex | capital-gain | capital-loss | hours-per-week | native-country || (Label) income-level|\n",
+    "|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:|:-:||:-:\n",
+    "| **1** | 25| 4 | 226802 | 1 | 7 | 4 | 7 | 3 | 2 | 1 | 0 | 0 | 40 | 38 || False (<50K) |\n",
+    "| **2** | 28| 2 | 336951 | 7 | 12 | 2 | 11 | 0 | 4 | 1 | 0 | 0 | 40 | 38 || True (>50K) |\n",
+    "\n",
+    "** Replace: **\n",
+    "* `VERSION` - with your version name, such as \"v1\"\n",
+    "* `MODEL_NAME` - with the model name you used above, such as \"scikit_learn_prediction\"\n",
+    "\n",
+    "Test the model with an online prediction using the data of a person who makes <50K."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "! gcloud ml-engine predict --model [MODEL_NAME] --version [VERSION] --json-instances less_than_50K.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Test the model with an online prediction using the data of a person who makes >50K."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "! gcloud ml-engine predict --model [MODEL_NAME] --version [VERSION] --json-instances more_than_50K.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Use Python to make online predictions\n",
+    "Test the model with the entire test set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "import googleapiclient.discovery\n",
```

Comment:
I don't think you should clear all cells before you push to GitHub. It would be good to show the output from a clean run on a fresh GCP project to show what the outputs would look like if someone doesn't feel like actually executing it.

Best practice would probably be to make a GCP project, run the notebook once, and then delete the project. That way, no compromising information gets leaked.

Reply:
+1

Reply:
Fixed